clients/pkg/promtail/targets/file/type_sniff.go

```diff
@@ -0,0 +1,24 @@
+package file
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"net/http"
+)
+
+var errBinaryContent = errors.New("content is binary")
+
+// checkIfBinary uses http.DetectContentType, which implements https://mimesniff.spec.whatwg.org/
+func checkIfBinary(r io.Reader) error {
+	reader := bufio.NewReader(r)
+
+	// make a best-effort attempt to read into the buffer
+	buf, _ := reader.Peek(512) //nolint:errcheck
+
+	if http.DetectContentType(buf) == "application/octet-stream" {
```

Comment:
my only thought here is are there more types we would want to exclude? 

Would it be better to allowlist here or denylist?

Reply:
It's a fair question.

I think we _could_ make this configurable, but I'd be interested in which cases you see it being put to use?

Using mime types here is a bit of a bastardisation because those are typically confined to http, but it was a cute hack to solve a problem we happen to have in common with http servers in terms of content negotiation

Reply:
More like, `DetectContentType` can also return:

```
application/pdf
application/postscript
image/x-icon
audio/mpeg
...
```

There are a few dozen options it looks like

What I'm wondering is, instead of just checking for `application/octet-stream`

Instead should we make sure the returned type starts with `text/plain`?

Reply:
Ah, you're totally right.
I think I may have to go back to the drawing board for this... well spotted! :eagle: :eye: